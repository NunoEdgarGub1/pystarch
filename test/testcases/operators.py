x01 = 'a' * 5
x02 = 'a' * 'b'
x03 = 5 * 'a'
x04 = [] * []
x05 = not True
x06 = not []
x07 = not 5
x08 = 5 * 5
x09 = 'a' - 'b'
x10 = 'a' + 'b'
x11 = [] + []
x12 = (1, 'a') + ('b', 2)
x13 = (1, 2) + (3, 4)
x14 = 'abc {0} {1} xyz' % (25, 50)
