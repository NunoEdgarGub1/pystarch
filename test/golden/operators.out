x01 Str aaaaa
x02 Unknown
x03 Str aaaaa
x04 Unknown
x05 Bool False
x06 Bool True
x07 Bool False
x08 Num 25
x09 Num
x10 Str ab
x11 List(Unknown) []
x12 Tuple(Num,Str,Str,Num) (1, 'a', 'b', 2)
x13 Tuple(Num,Num,Num,Num) (1, 2, 3, 4)
x14 Str

testcases/operators.py:2 type-error "Mult" ([Str, Str] vs {Num} or {Num, Str})
testcases/operators.py:4 type-error "Mult" ([List(Unknown), List(Unknown)] vs {Num} or {Num, Str})
testcases/operators.py:6 type-error "List" (List(Unknown) vs Bool)
testcases/operators.py:7 type-error "Num" (Num vs Bool)
testcases/operators.py:9 type-error "Str" (Str vs Num)
testcases/operators.py:9 type-error "Str" (Str vs Num)
